src/components/toolbar/helpers.ts

```diff
@@ -367,24 +405,28 @@ export const indentParagraph =
       sectionContent
     )
 
-    // Replace original paragraph and moved nodes with the new section
-    tr.replaceWith(beforeParagraph, sectionEnd, newSection)
+    // Set indentation metadata for track changes
+    tr.setMeta('indentation', {
+      type: 'indent',
+      nodeType: 'paragraph',
+    })
+
+    tr.delete(beforeParagraph, sectionEnd)
+    tr.insert(beforeParagraph, newSection)
 
     // Set selection inside the title of the new section
     tr.setSelection(TextSelection.create(tr.doc, beforeParagraph + 2))
```

Comment:
you will need to map that `beforeParagraph`

Reply:
Should be fixed.